Tests/JaegerTests/JaegerModelsTests.swift

```diff
@@ -0,0 +1,152 @@
+        XCTAssertEqual(jaegerLog.fields[1].vDouble, 42)
+        XCTAssertEqual(jaegerLog.timestamp, Int64(date.timeIntervalSince1970.microseconds))
+    }
+    
+    func testJaegerSpanReferenceChildOfConversion() {
+        let uuid = UUID()
+        let context =  Span.Context(traceId: uuid, spanId: uuid)
+        let ref = Span.Reference(refType: .childOf, context: context)
+        let jaegerRef = JaegerSpan.JaegerSpanReference(ref: ref)
+        
+        XCTAssertEqual(jaegerRef.spanId, Int64(bitPattern: uuid.firstHalfBits))
+        XCTAssertEqual(jaegerRef.traceIdLow, Int64(bitPattern: uuid.firstHalfBits))
+        XCTAssertEqual(jaegerRef.traceIdHigh, Int64(bitPattern: uuid.secondHalfBits))
+        XCTAssertEqual(jaegerRef.refType, .childOf)
+    }
+    
+    func testJaegerSpanReferenceFollowsFromConversion() {
+        let uuid = UUID()
+        let context =  Span.Context(traceId: uuid, spanId: uuid)
+        let ref = Span.Reference(refType: .followsFrom, context: context)
+        let jaegerRef = JaegerSpan.JaegerSpanReference(ref: ref)
+        
+        XCTAssertEqual(jaegerRef.spanId, Int64(bitPattern: uuid.firstHalfBits))
+        XCTAssertEqual(jaegerRef.traceIdLow, Int64(bitPattern: uuid.firstHalfBits))
+        XCTAssertEqual(jaegerRef.traceIdHigh, Int64(bitPattern: uuid.secondHalfBits))
+        XCTAssertEqual(jaegerRef.refType, .followsFrom)
+    }
+    
+    func testJaegerSpanConversion() {
+        let tracer = EmptyTestTracer()
+        let startTime = Date()
+        let endTime = Date().addingTimeInterval(1)
+        let tag = Tag(key: "testKey", tagType: .string("testType"))
+        let log = Log(timestamp: startTime, fields: [tag])
+        let uuid = UUID()
+        let name = "oppName"
+        let context =  Span.Context(traceId: uuid, spanId: uuid)
+        let ref = Span.Reference(refType: .childOf, context: context)
+
+        var span = Span(tracer: tracer,
+                        spanRef: context,
+                        parentSpanId: uuid,
+                        operationName: name,
+                        references: [ref],
+                        flag: .debug,
+                        startTime: startTime,
+                        tags: [tag.key: tag],
+                        logs: [log])
+        span.finish(at: endTime)
+        
+        let jaegerSpan = JaegerSpan(span: span)
+        
+        XCTAssertEqual(jaegerSpan.spanId, Int64(bitPattern: uuid.firstHalfBits))
+        XCTAssertEqual(jaegerSpan.traceIdLow, Int64(bitPattern: uuid.firstHalfBits))
+        XCTAssertEqual(jaegerSpan.traceIdHigh, Int64(bitPattern: uuid.secondHalfBits))
+        XCTAssertEqual(jaegerSpan.operationName, name)
+        XCTAssertEqual(jaegerSpan.flags, 2)
+        XCTAssertEqual(jaegerSpan.startTime, Int64(startTime.timeIntervalSince1970.microseconds))
+        XCTAssertEqual(jaegerSpan.incomplete, false)
+        XCTAssertEqual(jaegerSpan.tags![0].key, "testKey")
```

Comment:
can we find a more elegant way to express this than force unwrapping? If this fails, the whole test suite crashes